### ISSUE TITLE
Remove Documents from Collection no longer works

### ISSUE THREAD

**Issue by gpolonus** [tsgrp/HPI#1229]
Remove from collection is now part of the action framework; it expects a list of document IDs rather than a collection ID, therefore the condition evaluator implementation will need to be able to take a list of selected documents and derive the related collection.

**Comment by dgrumieaux** [comment]
Documents may be in many collections. The general contract for actions is that the ID passes in the object ID that is to be modified. Sounds like there will need to be more thought to pass in the collection name or ID as the action will not know which collection to remove the object from.

**Comment by gsteimer** [comment]
I was thinking that we'd do something like this:

- If the `collectionId` param is passed in and non-null, then do what the condition does now
- if the `objectId` param is passed in and non-null:
  - Get the attached collection ID based on the relationship
  - Cache the ID and use it in the evaluationCache key
  - Check the user's permissions on the collection

The current code uses this for the key: `EvaluationCache.HAS_WRITE_PERMISSION` which is probably way too generic and could be getting tripped up with other conditions.  I'm guessing the key should really be something like:

```java
String key = collectionId + EvaluationCache.HAS_WRITE_PERMISSION;
```

**Comment by dgrumieaux** [comment]
But how do you know which collection to remove the document/documents from?

**Comment by dgrumieaux** [comment]
Can a document be in only one collection at a time?

**Comment by gsteimer** [comment]
Good point - a document can be in more than one collection.  So maybe the fix for this would be to figure out how we can get the condition to actually evaluate on the current collection ID within the confines of the action/condition framework.

**Comment by dgrumieaux** [comment]
Maybe the collection model knows which collection is active on the page, and the HPI action can find the ID from that.

**Comment by gsteimer** [comment]
I realized that the `RemoveFromCollectionActionExecuter` will also fail once we get the action to appear in the list since both `objectIds` and `collectionId` are required parameters.

**Comment by gsteimer** [comment]
bumping this to 2.5 since now we can't 'fix' this by simply removing the condition.

**Comment by peterkk** [comment]
Closed with commit #16727
